bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TableProvider.java

```diff
@@ -0,0 +1,336 @@
+                        break;
+                    case TypeTags.ARRAY_TAG:
+                        BType elementType = ((BArrayType) param.getType()).getElementType();
+                        if (elementType.getTag() == TypeTags.BYTE_TAG) {
+                            byte[] blobData = params.getBytes();
+                            stmt.setBlob(index, new ByteArrayInputStream(blobData), blobData.length);
+                        } else {
+                            Object[] arrayData = TableUtils.getArrayData((ArrayValue) param);
+                            stmt.setObject(index, arrayData);
+                        }
+                        break;
+                }
+            }
+            stmt.executeUpdate();
+        } catch (SQLException e) {
+            throw new BallerinaException(
+                    "error in executing statement : " + queryStatement + " error:" + e.getMessage());
+        } finally {
+            releaseResources(conn, stmt);
+        }
+    }
+
+    private void prepareAndExecuteStatement(String queryStatement, MapValue<String, Object> constrainedType) {
+        PreparedStatement stmt = null;
+        Connection conn = this.getConnection();
+        try {
+            stmt = conn.prepareStatement(queryStatement);
+            TableUtils.prepareAndExecuteStatement(stmt, constrainedType);
+        } catch (SQLException e) {
+            throw new BallerinaException(
+                    "error in executing statement : " + queryStatement + " error:" + e.getMessage());
+        } finally {
+            releaseResources(conn, stmt);
+        }
+    }
+
+    private void releaseResources(Connection conn, Statement stmt) {
+        try {
+            if (stmt != null) {
+                stmt.close();
+            }
+        } catch (SQLException e) {
+            throw new BallerinaException("error in releasing table statement resource : " + e.getMessage());
+        }
+        try {
+            if (conn != null && !conn.isClosed()) {
+                conn.close();
+            }
+        } catch (SQLException e) {
+            throw new BallerinaException("error in releasing table connection resource : " + e.getMessage());
+        }
+    }
+
+    public int getRowCount(String tableName) {
+        Statement stmt = null;
+        Connection conn = this.getConnection();
+        try {
+            stmt = conn.createStatement();
+            ResultSet rs = stmt.executeQuery(TableConstants.SQL_COUNT + tableName);
+            int rowCount = 0;
+            if (rs.next()) {
+                rowCount = rs.getInt(1);
+            }
+            return rowCount;
+        } catch (SQLException e) {
+            throw new BallerinaException("error in executing statement to get the count : " + stmt + " error:"
+                                         + e.getMessage());
+        } finally {
+            releaseResources(conn, stmt);
+        }
+    }
```

Comment:
Shall we move all the above classes to a relevant sub package? for eg. `TableIterator.java` is part of the Table value, hence can move it to `org.ballerinalang.jvm.values` package.

Reply:
fixed